### PR TITLE
Validate tags

### PR DESCRIPTION
**Pull request by SteveHNH**
Catch invalid YAML issues related to tags.conf

We don't die if this fails, but we do throw in a warning for it.
Validate will now check that `tags.conf` is v alid as well

### CODE REVIEW

File: insights/client/config.py [hunk 1]
```diff
@@ -242,6 +242,10 @@
         # non-CLI
         'default': os.path.join(constants.default_conf_dir, 'remove.conf')
     },
+    'tags_file': {
+        # non-CLI
+        'default': os.path.join(constants.default_conf_dir, 'tags.conf')
```
Comment (gravitypriest) [inline insights/client/config.py:247]: should be `tags.yaml`

File: insights/client/constants.py [hunk 2]
```diff
@@ -10,6 +10,7 @@ class InsightsConstants(object):
     command_blacklist = ('rm', 'kill', 'reboot', 'shutdown')
     default_conf_dir = os.getenv('INSIGHTS_CONF_DIR', default='/etc/insights-client')
     default_conf_file = os.path.join(default_conf_dir, 'insights-client.conf')
+    default_tags_file = os.path.join(default_conf_dir, 'tags.conf')
```
Comment (gravitypriest) [inline insights/client/constants.py:13]: should be `tags.yaml` :)

File: insights/client/collection_rules.py [hunk 3]
```diff
@@ -408,10 +409,25 @@ def get_rm_conf(self):
...
+        Try to load the tags.conf file
+        '''
+        if not os.path.isfile(self.tags_file):
+            logger.info("%s does not exist", self.tags_file)
+            return None
+        else:
+            try:
+                load_yaml(self.tags_file)
+            except RuntimeError as e:
+                logger.info(e)
```
Comment (gravitypriest) [inline insights/client/collection_rules.py:423]: maybe `logger.warning` in this function to match remove.conf validation errors?

File: insights/client/config.py (hunk 1 above)
Comment (SteveHNH) [reply insights/client/config.py:247]: ah! didn't realize we had made that switch. On it

**Comment by SteveHNH** [comment]
test this please

**Comment by koalakangaroo** [comment]
@SteveHNH - May I know what the tags.conf file is used for in the client?

**Comment by SteveHNH** [comment]
@koalakangaroo Tags are used to give machines more searchable fields in inventory, I believe. I think the main use right now is for groups. Multiple uploaders will submitting tags. This is the client implementation of it. Satellite can do it as well.

**Pull request merged**